tests/ceph_volume/test_ceph_volume_redeploy.py

```diff
@@ -0,0 +1,127 @@
+from json import loads
+
+from ceph.waiter import WaitUntil
+from cephci.utils.configs import get_cloud_credentials, get_configs
+from cli.cephadm.cephadm import CephAdm
+from cli.cloudproviders import CloudProvider
+from cli.cluster.node import Node
+from cli.cluster.volume import Volume
+from cli.ops.cephadm import generate_bootstrap_config
+from cli.utilities.utils import get_lvm_on_osd_container, get_running_containers
+from utility.log import Log
+
+log = Log(__name__)
+
+
+class RemoveCephVolumeError(Exception):
+    pass
+
+
+def _create_volume_in_bulk(nodes, cloud, size, count):
+    """Create volume in bulk and attach to node
+    Args:
+        nodes (ceph): nodes on volumes to create
+        cloud (cloudproviders): cloud provider obj
+        size (str): size of volume
+        count (int): number of volume
+    """
+    nodes = nodes if type(nodes) in (list, tuple) else [nodes]
+    for node in nodes:
+        for i in range(count):
+            volname = f"{node.ceph_nodename}-vol-{size}-{i}"
+            Volume(volname, cloud).create(size=size)
+            Node(node.ceph_nodename, cloud).attach_volume(volname)
+    return True
+
+
+def _get_osd_db_id(osd_ids, lvm_list):
+    """Get OSD id and associated devices name
+    Args:
+        osd_ids (list): list of osd ids
+        lvm_list (list): list of lvm
+    return (dict): osd id associated with device
+    """
+    osds = {}
+    for id in osd_ids:
+        if lvm_list.get(id, {})[0].get("type", {}) == "db":
+            osds[id] = lvm_list.get(id, {})[0].get("devices", {})
+    return osds
+
+
+def run(ceph_cluster, **kw):
+    """Re-deploy non-collocated OSDs with wrong dedicated DB size"""
+
+    # Get configs
+    get_configs()
+    config = kw.get("config")
+
+    # Get the installer and OSD nodes
+    installer = ceph_cluster.get_nodes(role="installer")[0]
+    osd_nodes = ceph_cluster.get_nodes(role="osd")
+
+    # Get device spec
+    device_spec = config.get("spec", {}).get("spec", {})[0].get("spec", {})
+
+    # Get cloud creadentials
+    cloud_type = config.get("cloud-type")
```

Comment:
These variables can be moved to method `_create_volume_in_bulk`

Reply:
Done, updated.